ksqldb-rest-app/src/main/java/io/confluent/ksql/rest/server/execution/ConnectExecutor.java

```diff
@@ -54,8 +59,7 @@ public static StatementExecutorResponse execute(
     final List<String> errors = validate(createConnector, client);
     if (!errors.isEmpty()) {
       final String errorMessage = "Validation error: " + String.join("\n", errors);
-      return StatementExecutorResponse.handled(Optional.of(new ErrorEntity(
-          statement.getStatementText(), errorMessage)));
+      throw new KsqlException(errorMessage);
```

Comment:
Assuming client connection failed here thus KsqlException instead of KsqlRestException?

Reply:
Not quite -- KsqlException means it's a user error, which is necessarily the case if there's a validation error but not necessarily the case if other errors are encountered.

The final exception type returned from the engine is always KsqlRestException. KsqlException is automatically wrapped in KsqlRestException here: https://github.com/confluentinc/ksql/blob/7b3fb6b078a02c48ab4fe976ca836b8efe514a59/ksqldb-rest-app/src/main/java/io/confluent/ksql/rest/server/resources/KsqlResource.java#L328-L341

For the non-validation connector errors, we don't know whether it's a user error or not without inspecting the error itself, which is why this PR simply returns a KsqlRestException directly rather than trying to figure out what inner error to throw in those cases.